### ISSUE TITLE
dynamic require breaks browserify builds

### ISSUE THREAD

**Issue by austinfrey** [JoinColony/colonyJS#468]
## Description

Browserify does not deal with dynamic `require` module loading. Specifically when loading the contracts here: https://github.com/JoinColony/colonyJS/blob/65ae6abf63dd79380ec04f4b4c5e64aba77e2883/packages/colony-js-contract-loader-network/src/NetworkLoader.js#L21

It looks like all the contracts are loaded from the contracts folder so would it be possible explicitly require the contracts, or is it dynamic for other reason maybe? 

Something like this works for me but unsure if it would be problematic for other reason.

create a new file that explicitly requires all the contracts

*colony-js-contract-loader-network/contracts/index.js*
```js
module.exports = {
    1: {
        EtherRouter: require('./versioned/v1/EtherRouter.json'),
        // and so on
    },
    2: {
        EtherRouter: require('./versioned/v1/EtherRouter.json'),
        // and so on
    }, 
    3: {
        EtherRouter: require('./versioned/v1/EtherRouter.json'),
        // and so on
    },
    4: {
        EtherRouter: require('./versioned/v1/EtherRouter.json'),
        // and so on
    }
}
```

and then in colony-js-contract-loader-network/src/NetworkLoader.js just require all the contracts

*colony-js-contract-loader-network/src/NetworkLoader.js*
```js
import assert from 'assert';
import ContractLoader, {
  truffleTransform,
} from '@colony/colony-js-contract-loader';

import type { RequiredContractProps } from '@colony/colony-js-contract-loader';

const NETWORKS = {
  GOERLI: 'goerli',
  MAINNET: 'mainnet',
};

...

var VERSIONED_CONTRACTS = require('../contracts')

class NetworkLoader extends ContractLoader {
  _network: ?Network;

  constructor({ network = DEFAULT_NETWORK }: { network: Network } = {}) {
    super({ transform: truffleTransform });
    this._network = network;
  }

...

```

- [x] TODO write description

**Comment by chmanie** [comment]
Hey @austinfrey! Thank you for the detailed error descriptions. ColonyJS is undergoing a major rewrite right now (for version 2.0). I would like to invite you to test the new version which is `@colony/colony-js@2.0.0-alpha.1`

You can find the code [here](https://github.com/JoinColony/colonyJS/tree/v2) and the documentation with a simple example here: https://joincolony.github.io/colonyJS/index.html

There might be some issues still, but we would rather focus our resources on improving and supporting the new version rather than providing support on the old one. Hope you understand.

Don't hesitate to ask questions here in GitHub - or join our [discord](https://discord.gg/PDEQvrC) direct messages :)

**Comment by austinfrey** [comment]
@chmanie thanks for the reply, i think that is fine for this particular issue so thanks for the links! for my other issue #467, i feel like this is something that should be fixed since this is the package that is downloaded when following any install instructions in the docs as well as when using the colonyStarter boilerplates. 

i'm happy to use my patched version for now , but it could be a pain point for anyone else trying to get started.

**Comment by chmanie** [comment]
Hey! I have just released `@colony/colony-js@2.0.0` which should not have this issue. Check out the docs here:

https://joincolony.github.io/colonyJS/

**Comment by austinfrey** [comment]
@chmanie thanks for the ping. I'll check it out!
